s3/pkg/service/object.go

```diff
@@ -630,8 +630,9 @@ func (s *s3Service) CopyObject(ctx context.Context, in *pb.CopyObjectRequest, ou
 	targetObject.StorageMeta = res.Meta
 	targetObject.Location = targetBackendName
 	targetObject.TenantId = tenantId
+	targetObject.UserId = srcBucket.UserId
 	// this is the default acl setting
-	targetObject.Acl = in.Acl
+	targetObject.Acl = targetBucket.Acl
```

Comment:
Hi @Click2cloud-Kukekar , why use targetObject.Acl as the targetBucket.Acl, IMO, user can specify Acl in the copy request.

Reply:
@sfzeng Okay, but are we giving any option (on dashboard) to user to add ACL details while copy object operation. If yes, then we have to put 1 condition when this function will be used for migration purpose.
During migration in.ACL was `null` which create issue while downloading object from target bucket as at ACL checkpoint it gets `null` value.

Reply:
On dashboard there is no such option, but it can be used through REST request directly, so can we set it to be targetBucket.Acl if in.ACL is not set, as it's the default config, and if in.ACL is set then we use it?

Reply:
@sfzeng Made changes as suggested, please review.
Thanks